lib/devise.rb

```diff
@@ -325,6 +325,9 @@ def self.add_mapping(resource, options)
     mapping = Devise::Mapping.new(resource, options)
     @@mappings[mapping.name] = mapping
     @@default_scope ||= mapping.name
+    if defined? Rails and Rails.try(:application).try(:config).try(:api_only)
+      @@helpers = []
```

Comment:
I don't fully understand why this is needed - but by default @@helpers is a Set - so, if it is necessary to clear it, it's probably better to set to Set.new rather than empty array